src/flb_pack.c

```diff
@@ -923,13 +924,23 @@ flb_sds_t flb_pack_msgpack_to_json_format(const char *data, uint64_t bytes,
             case FLB_PACK_JSON_DATE_JAVA_SQL_TIMESTAMP:
             /* Format the time, use microsecond precision not nanoseconds */
                 gmtime_r(&tms.tm.tv_sec, &tm);
-                s = strftime(time_formatted, sizeof(time_formatted) - 1,
+                s = strftime(time_formatted, sizeof(time_formatted),
                              FLB_PACK_JSON_DATE_JAVA_SQL_TIMESTAMP_FMT, &tm);
 
+                max_len = sizeof(time_formatted) - s;
                 len = snprintf(time_formatted + s,
-                               sizeof(time_formatted) - 1 - s,
+                               max_len,
                                ".%06" PRIu64,
```

Comment:
I think there's an error in the worst case scenario and this `snprintf` could use up to 21 characters (leading dot + 20 characters which is the longest number a `uint64_t` can hold). Additionally, the `snprintf` in line 954 could use up to 22 characters (leading dot, uint64 and trailing T).

So considering we're not so constrained when it comes to memory, I'd make `time_formatted` at least 40 characters long.

Reply:
Yeah, I know it shouldn't but considering the context obsessing over a few stack bytes makes no sense to me. 
However, it's not a huge concern so it can stay like that.